runtime/rasdump/javadump.cpp

```diff
@@ -1484,6 +1484,87 @@ JavaCoreDumpWriter::writeEnvUserArgsHelper(J9VMInitArgs *vmArgs)
 				_OutputStream.writeCharacters("\n");
 			}
 		}
+
+		/* Ignored options for -Xjit and -Xaot follows the following rules:
+		 * 1. If Xjit is followed by -Xint or -Xnojit,
+			then any -Xjit:optionString beforehand appears as ignored
+		 * 2. If XX:+MergeCompilerOptions is present,
+			then all Xjit options after Xint or Xnojit should not be ignored
+		 * 3. If multiple Xjit string appear, all but the last one will appear as ignored
+		 *
+		 * and the same for -Xaot
+		 */
+
+		int numXjit = 0, numXaot = 0;
+		bool hasXXMerge = false;
+
+		/* These indices ensure all the options before them will be printed as ignored */
+		int ignoreXjitBefore = 0, ignoreXaotBefore = 0;
+
+		for (jint i = 0; i < args->nOptions; i++) {
+			const char * optionString = args->options[i].optionString;
+
+			if (0 == strncmp(optionString, "-Xint", 5)) {
+				/* Case 1: ignores all options before */
+				ignoreXjitBefore = numXjit;
+				ignoreXaotBefore = numXaot;
+			} else if (0 == strncmp(optionString, "-Xnojit", 7)) {
+				/* Case 1: ignore all jit options before */
+				ignoreXjitBefore = numXjit;
+			} else if (0 == strncmp(optionString, "-Xnoaot", 7)) {
+				/* Case 1: ignore all aot options before */
+				ignoreXaotBefore = numXaot;
+			} else if (0 == strncmp(optionString, "-XX:+MergeCompilerOptions", 25)) {
```

Comment:
To be pedantic I would also check whether `-XX:-MergeCompilerOptions` is specified after `-XX:+MergeCompilerOptions`, invalidating it.